src/Confluent.Kafka/Producer.cs

```diff
@@ -284,6 +271,26 @@ public Producer(IEnumerable<KeyValuePair<string, object>> config, bool manualPol
                 .ToList()
                 .ForEach((kvp) => { configHandle.Set(kvp.Key, kvp.Value.ToString()); });
 
+            // Note: setting default topic configuration properties via default.topic.config is depreciated 
+            // and this functionality will be removed in a future version of the library.
+            var defaultTopicConfig = (IEnumerable<KeyValuePair<string, object>>)config.FirstOrDefault(prop => prop.Key == "default.topic.config").Value;
+            if (defaultTopicConfig != null)
+            {
+                defaultTopicConfig.ToList().ForEach(
+                    (kvp) => { configHandle.Set(kvp.Key, kvp.Value.ToString()); }
+                );
+            }
+
+            // Note: changing the default value of produce.offset.report at the binding level is less than
+            // ideal since it means the librdkafka configuration docs will no longer completely match the 
+            // .NET client. The default should probably be changed in librdkafka as well.
+            if (config.FirstOrDefault(prop => prop.Key == "produce.offset.report").Value == null &&
+                (defaultTopicConfig == null ||
+                 defaultTopicConfig.FirstOrDefault(prop => prop.Key == "produce.offset.report").Value == null))
+            {
+                configHandle.Set("produce.offset.report", "true");
```

Comment:
Isn't it easier to just set this (and other defaults) at the creation of the configHandle?

Inverse priority order:
 * librdkafka default settings (produce.offset.report=false)
 * dotnet default settings (produce.offset.report=true)  <== set here
 * global explicit config
 * default.topic.config explicit config

Reply:
Not 100% sure I understand exactly what you're saying, but I simplified as a result of your comment. A modifiedConfig is now prepared, then all properties set in one loop (this is logically simpler).